src/components/transaction-dialog/transaction-dialog.js

```diff
@@ -0,0 +1,66 @@
+import angular from "angular";
+import template from "./transaction-dialog.html";
+
+import ngAnimate from "angular-animate";
+import ngMaterial from "angular-material";
+
+export class TransactionDialogController {
+  constructor($mdDialog, $filter) {
+    this.$mdDialog = $mdDialog;
+    this.$filter = $filter;
+    this.tags = ["transport", "food", "salary"];
+    this.accounts = ["Privat", "BoaBank", "Cash"];
+  }
+
+  $onChanges(changes) {
+    if (changes.isOpen.currentValue) {
+      this.showDialog(changes.isOpen.currentValue.event);
+    }
+  }
+
+  showDialog(ev) {
+    this.$mdDialog.show({
+      template,
+      controllerAs: "$ctrl",
+      controller: () => new DialogController(this),
```

Comment:
* you do not need to instantiate class here -- you just need to pass class 
* you need to use `locals` prop in show() to pass variables that you need to controller

Reply:
if I use locals - it is the same, as I use this as props. here is an example
![image](https://user-images.githubusercontent.com/14272396/38206870-b2b62c40-36b4-11e8-8deb-4869d7b7b95a.png)

Reply:
What do you mean "to pass class"?